modules/kubernetes-addons/argocd/argocd-application/helm/templates/application.yaml

```diff
@@ -11,11 +11,16 @@ spec:
     repoURL: {{ .Values.source.repoUrl }}
     targetRevision: {{ .Values.source.targetRevision }}
     path: {{ .Values.source.path }}
-    helm:
-      valueFiles:
-        - 'secrets://values.yaml'
-        - 'secrets://values.ci.yaml'
-      values: {{ toYaml .Values.source.helm.values | toString | indent 6 }}
+    {{ if ((.Values.source).plugin).name -}}
+      plugin:
+        name: {{ .Values.source.plugin.name }}
+    {{ else if ((.Values.source.helm).values) -}}
+      helm:
+        valueFiles:
+          - 'secrets://values.yaml'
+          - 'secrets://values.ci.yaml'
```

Comment:
remove the `secrets://` prefix. it should not be done.
what it does is that it decrypts all the app manifests. Since this is a root app, it should not be used.

otherwise - all apps generated by the root app (app of apps pattern) will have all secret values in open text that you can see in the argocd gui.

That was my  early mistake, as I didn't properly understood how **secrets://** prefix worked back in the days.